homeassistant/components/xiaomi_miio/config_flow.py

```diff
@@ -47,36 +50,76 @@ async def async_step_user(self, user_input=None):
             step_id="user", data_schema=CONFIG_SCHEMA, errors=errors
         )
 
+    async def async_step_zeroconf(self, user_input=None):
+        """Handle zeroconf discovery."""
+        if user_input is None:
+            return self.async_abort(reason="not_xiaomi_miio")
+
+        name = user_input.get("name")
+        self.host = user_input.get("host")
+        mac_address = user_input.get("properties", {}).get("mac")
+
+        if not name or not self.host or not mac_address:
+            return self.async_abort(reason="not_xiaomi_miio")
+
+        # format mac (include semicolns and make uppercase)
+        if len(mac_address) == 12:
+            mac_address = ":".join(mac_address[i : i + 2] for i in range(0, 12, 2))
```

Comment:
Why not use `format_mac` as defined in `helpers/device_registry.py` ?

Reply:
Thank you for the hint